lib/collection/property.js

```diff
@@ -178,12 +187,12 @@ _.extend(Property, /** @lends Property */ {
         // extract all arguments to get the list of variables
         !_.isArray(variables) && (variables = _.tail(arguments));
 
-        return str.replace(this.REGEX_EXTRACT_VARS, function (match, token) {
-            var r = _.findValue(variables, token);
-            r && _.isFunction(r) && (r = r());
-            r && _.isFunction(r.toString) && (r = r.toString());
-            return nativeTypes[(typeof r)] ? r : match;
-        });
+        while (resolvedStr != str && ++loopCounter < 20) {
```

Comment:
The original function was written with recursion in mind. Any particular reason you're using iterations?